### PR TITLE
docs: release process

### PR DESCRIPTION
**Pull request by pameladelgado**
Documents the release process for Renku, including rollout to production deployments.

### CODE REVIEW

**Comment by jachro** [comment]
I've just noticed there's also https://github.com/SwissDataScienceCenter/renku/blob/release-docs/RELEASE-NOTES.md file which seems to be a bit outdated (e.g. mentions `.travis.yaml` file). Do we actually want to keep it?

File: docs/releases.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+.. _releases:
+
+Releases
+=========
+
+Next planned releases
+---------------------
+
+* Sprint June 10 - June 30 2022 release: July 5th
```
Comment (rokroskar) [inline docs/releases.rst:9]: this will go stale immediately, i.e. it already has. I suggest we remove dates from here

**Review by rokroskar — COMMENTED** [review]
I'm not sure the release process belongs in the general renku docs tbh. It's not really a generic feature of the project but rather a specific detail of how our team works.

File: docs/releases.rst [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+in a section above the one for the most recent release.
+The auto-PR is only merged with an updated changelog/release notes.
+Ideally, components would release when there are meaningful changes or so
+there's a release at least every 3 weeks.
+
+If two components need to bee coordinated for a release (e.g. new version
+of Graph needs new version of RP), they should combine their auto-PRs and
+merge them in a single go. Generally, components should strive to keep
+dependencies between them to a minimum and coordinated component releases
+should not usually be needed(by using APi versioning, feature flags etc.).
```
Comment (rokroskar) [inline docs/releases.rst:31]:
```suggestion
should not usually be needed (by using APi versioning, feature flags etc.).

```

File: docs/releases.rst [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+information.
+
+Pre-requisites [Graph, UI, RP]
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Components release whenever they want, the schedule is up to them. They are
+responsible for updating the changelog in the auto-PR in the renku repository,
+in a section above the one for the most recent release.
+The auto-PR is only merged with an updated changelog/release notes.
+Ideally, components would release when there are meaningful changes or so
```
Comment (rokroskar) [inline docs/releases.rst:24]: This sentence is confusing. I think you can just leave it at "the schedule is up to them"

File: docs/releases.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+accidentally included in a release. A new auto-PR with the fix can then be
+created. The main release DOES NOT wait for a fix.
+
+If a regression made it into a release, the fix must add an acceptance test
+that would catch it, within reason.
+
+Process [YAT]
+~~~~~~~~~~~~~
+
+* Make a Pull Request in renku repository with updates relevant -and adapted- to USERS and ADMINS in the CHANGELOG.md file. Ask ORDES for review when necessary.
```
Comment (rokroskar) [inline docs/releases.rst:44]:
```suggestion
* Make a Pull Request in renku repository with updates relevant -and adapted- to USERS and ADMINS in the CHANGELOG.md file. 
```

File: docs/releases.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+* Plan a maintenance, create it in statuspage for all Renku deployments managed by us. Coordinate with external Renku deployments admins (UniFR, SV).
+
+    * Limited should be rolled out at least 5 days before the Renkulab rollout, to give a change to find bugs early.
+
+* Make the tag in renku repository, follow process of chart publishing and deployment on staging.
+
+* Merge terraform-renku PR, monitor Flux upgrading renku everywhere.
+
+* If relevant, notify users on discourse, twitter using visual tools and short sentences. Coordinate with ORDES.
```
Comment (rokroskar) [inline docs/releases.rst:60]:
```suggestion
* If relevant, notify users on discourse, twitter using visual tools and short sentences. 

```

File: docs/releases.rst [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+that would catch it, within reason.
+
+Process [YAT]
+~~~~~~~~~~~~~
+
+* Make a Pull Request in renku repository with updates relevant -and adapted- to USERS and ADMINS in the CHANGELOG.md file. Ask ORDES for review when necessary.
+
+* Rollout updates
+
+    * For non-breaking changes: create two pull requests in terraform-renku repository with the changes, one for limited and one for renkulab. Important: add the scheduled maintenance label to help avoiding it being merged before time.
```
Comment (rokroskar) [inline docs/releases.rst:48]: There is one PR per deployment - but we should say here that limited can be used as a "canary" and others follow ~1 week later

**Review by rokroskar — CHANGES_REQUESTED** [review]
Nice, it's getting there :)

**Review by rokroskar — APPROVED** [review]

**Pull request merged**